geode-for-redis/src/main/java/org/apache/geode/redis/internal/data/RedisList.java

```diff
@@ -168,8 +195,10 @@ public void applyAddByteArrayDelta(byte[] bytes) {
 
   @Override
   public void applyRemoveElementsByIndex(List<Integer> indexes) {
-    for (int index : indexes) {
-      elementRemove(index);
+    if (indexes.size() == 1) {
+      elementRemove(indexes.get(0));
+    } else {
+      elementList.removeIndexes(indexes);
```

Comment:
I think this line needs to be synchronized, to prevent concurrent modification/reading of the contents of `elementList` by other methods. For consistency, maybe `elementList.removeIndexes()` could be wrapped in a synchronized method in `RedisList`